### PR TITLE
Snaps improvements

### PR DESCRIPTION
**Pull request by Montoya**
This adds more resources in the snaps development guide and corrects a typo in the snaps introduction guide.

### CODE REVIEW

File: docs/guide/snaps.md [hunk 1]
```diff
@@ -153,7 +153,7 @@ export const onRpcRequest: OnRpcRequestHandler = ({ origin, request }) => {
 };
 ```
 
-Modify the text in the `description` or `textAreaContent` field. Refresh the dapp in your browser, and click the **Connect** button to reinstall the snap.
+Modify the text in the `description` or `textAreaContent` field. Refresh the dapp in your browser, and click the **Reconnect** button to reinstall the snap.
```
Comment (FrederikBolding) [inline docs/guide/snaps.md:156]:
```suggestion
Modify the text in the `description` or `textAreaContent` field and click the **Reconnect** button to reinstall the snap.
```
I don't think refreshing should be necessary?

File: docs/guide/snaps-development-guide.md [hunk 2]
```diff
@@ -406,15 +406,20 @@ You can review the growing number of [example snaps](https://github.com/MetaMask
...
+- [Template Snap](https://github.com/MetaMask/template-snap-monorepo) - A template that includes TypeScript/React and vanilla JS options and a CLI for building, packaging, and deploying your snap and a companion dapp
+- [Snaps Truffle Box](https://trufflesuite.com/boxes/metamask-snap-box/) - A template that combines the TypeScript template snap and Truffle so you can easily test snaps that use smart contracts with Ganache
+- [Test Snaps](https://github.com/MetaMask/test-snaps) - A collection of test snaps and a dapp for evaluating them
+- [Snaps Inspector](https://inspector.open-rpc.org/?request[jsonrpc]=2.0&request[method]=confirm&request[params][0]=hello&request[params][1]=more&request[params][2]=lorem%20ipsum&request[id]=0&url=npm:@metamask/test-snap-confirm&customTransport[type]=plugin&customTransport[name]=Snaps&customTransport[transport][type]=postmessageiframe&customTransport[transport][name]=PostMessageIframe&customTransport[uri]=https://xops.github.io/inspector-snaps-transport/) - An API tool to make JSON-RPC requests directly to Snaps
```
Comment (FrederikBolding) [inline docs/guide/snaps-development-guide.md:422]: Does this still work? 😅

Comment (Montoya) [reply docs/guide/snaps-development-guide.md:422]: No I think it's broken...

**Comment by Montoya** [comment]
This is ready for review again!

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**